File: backend/schemas/recommendation.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from uuid import UUID
from schemas.place import AmenityResponse


class RecommendationBase(BaseModel):
    place_name: str
    recommendation_text: str
    maps_link: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None


class RecommendationCreate(RecommendationBase):
    place_type_key: str
    amenity_keys: List[str] = []


class RecommendationUpdate(BaseModel):
    status: Optional[str] = None
    review_notes: Optional[str] = None


class RecommendationResponse(RecommendationBase):
    id: UUID
    user_id: Optional[UUID] = None
    place_type_key: str
    place_type_display_name: str
    place_type_icon: Optional[str] = None
    status: str
    created_at: datetime
    reviewed_at: Optional[datetime] = None
    reviewed_by: Optional[UUID] = None
    review_notes: Optional[str] = None
    amenities: List[AmenityResponse] = []
    
    class Config:
        from_attributes = True


class RecommendationListResponse(BaseModel):
    recommendations: List[RecommendationResponse]
    total: int
    page: int
    page_size: int
    total_pages: int

File: frontend/src/types/recommendation.ts
export interface RecommendationCreate {
  place_name: string;
  place_type_key: string;
  recommendation_text: string;
  maps_link?: string;
  address?: string;
  phone?: string;
  amenity_keys: string[];
}

export interface RecommendationResponse {
  id: string;
  user_id?: string;
  place_name: string;
  place_type_key: string;
  place_type_display_name: string;
  place_type_icon?: string;
  recommendation_text: string;
  maps_link?: string;
  address?: string;
  phone?: string;
  status: string;
  created_at: string;
  reviewed_at?: string;
  reviewed_by?: string;
  review_notes?: string;
  amenities: Array<{
    id: number;
    amenity_key: string;
    display_name: string;
    icon?: string;
  }>;
}

